src/components/pages/blocks/fetch-blocks.ts

```diff
@@ -0,0 +1,55 @@
+import { l2PublicClient } from "@/lib/chains";
+import { prisma } from "@/lib/prisma";
+import { fromViemBlock, Block } from "@/lib/types";
+import { range } from "lodash";
+
+const fetchLatestBlocksFromJsonRpc = async (
+  start: bigint,
+): Promise<Block[]> => {
+  const blocksPerPage = BigInt(process.env.NEXT_PUBLIC_BLOCKS_PER_PAGE);
+  const blocks = await Promise.all(
+    range(Number(start), Math.max(Number(start - blocksPerPage), -1)).map((i) =>
+      l2PublicClient.getBlock({ blockNumber: BigInt(i) }),
+    ),
+  );
+  return blocks.map(fromViemBlock);
+};
+
+const fetchLatestBlocksFromDatabase = async (
+  start: bigint,
+): Promise<Block[]> => {
+  const blocksPerPage = BigInt(process.env.NEXT_PUBLIC_BLOCKS_PER_PAGE);
+  const blocks = await prisma.block.findMany({
```

Comment:
Please use [this pattern](https://www.prisma.io/docs/orm/prisma-client/queries/relation-queries#select-specific-fields-of-included-relations) to also fetch the blocks transactions hashes, this is needed to display the correct number of transactions for each blocks.

Reply:
i stil used the findMany but now ive fixed it and it shows the correct number of transactions for each block